service/frontend/workflow_handler.go

```diff
@@ -4334,6 +4334,16 @@ func (wh *WorkflowHandler) validateWorkflowCompletionCallbacks(
 		)
 	}
 
+	if len(callbacks) > wh.config.MaxCallbacksPerWorkflow(ns.String()) {
```

Comment:
In the future we'll have alternative ways to attach callbacks to workflows (attach to a running workflow / attach to a workflow update).
Additional validation will be required in mutable state but this is a good enough starting point.